### DISCUSSION TITLE
issues about airflow_pod_template.yml

### DISCUSSION THREAD

**Discussion by chankamlam** [apache/airflow#18958] [Q&A]
My airflow info:
=================================================
Apache Airflow
version                | 2.2.0
executor               | KubernetesExecutor
task_logging_handler   | airflow.utils.log.file_task_handler.FileTaskHandler
sql_alchemy_conn       | postgresql://postgres:Ken5201314@192.168.31.198:5432/airflow3
dags_folder            | /home/vagrant/airflow/dags
plugins_folder         | /home/vagrant/airflow/plugins
base_log_folder        | /home/vagrant/airflow/logs
remote_base_log_folder |

System info
OS              | Linux
architecture    | x86_64
uname           | uname_result(system='Linux', node='Airflow2', release='4.15.0-154-generic', version='#161-Ubuntu SMP Fri Jul 30 13:04:17 UTC 2021', machine='x86_64', processor='x86_64')
locale          | ('en_US', 'UTF-8')
python_version  | 3.6.9 (default, Jan 26 2021, 15:33:00)  [GCC 8.4.0]
python_location | /usr/bin/python3

Tools info
git             | git version 2.17.1
ssh             | OpenSSH_7.6p1 Ubuntu-4ubuntu0.5, OpenSSL 1.0.2n  7 Dec 2017
kubectl         | Client Version: v1.22.2
gcloud          | NOT AVAILABLE
cloud_sql_proxy | NOT AVAILABLE
mysql           | NOT AVAILABLE
sqlite3         | NOT AVAILABLE
psql            | NOT AVAILABLE

Paths info
airflow_home    | /home/vagrant/airflow
system_path     | /home/vagrant/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
python_path     | /home/vagrant/.local/bin:/usr/lib/python36.zip:/usr/lib/python3.6:/usr/lib/python3.6/lib-dynload:/home/vagrant/.local/lib/python3.6/site-packages:/usr/local/lib/python3.6/dist-
                | packages:/usr/lib/python3/dist-packages:/home/vagrant/airflow/dags:/home/vagrant/airflow/config:/home/vagrant/airflow/plugins
airflow_on_path | True

Providers info
apache-airflow-providers-cncf-kubernetes | 2.0.3
apache-airflow-providers-ftp             | 2.0.1
apache-airflow-providers-http            | 2.0.1
apache-airflow-providers-imap            | 2.0.1
apache-airflow-providers-sqlite          | 2.0.1

My changing of airflow.cfg
======================================================

executor = KubernetesExecutor
sql_alchemy_conn = postgresql://postgres:Ken5201314@192.168.31.198:5432/airflow3
in_cluster = False
pod_template_file = airflow_pod_template.yml

My airflow_pod_template.yml:
======================================================
apiVersion: v1
kind: Pod
metadata:
  name: dummy-name
spec:
  initContainers:
    - name: git-sync
      image: k8s.gcr.io/git-sync/git-sync:v3.3.0
      env:
        - name: GIT_SYNC_BRANCH
          value: v2-1-stable
        - name: GIT_SYNC_REPO
          value: https://github.com/apache/airflow.git
        - name: GIT_SYNC_DEPTH
          value: 1
        - name: GIT_SYNC_ROOT
          value: /git
        - name: GIT_SYNC_DEST
          value: repo
        - name: GIT_SYNC_ADD_USER
          value: true
        - name: GIT_SYNC_ONE_TIME
          value: true
      volumeMounts:
        - name: airflow-dags
          mountPath: /git
  containers:
    - args: []
      command: []
      env:
        - name: AIRFLOW__CORE__EXECUTOR
          value: LocalExecutor
        # Hard Coded Airflow Envs
        - name: AIRFLOW__CORE__FERNET_KEY
          valueFrom:
            secretKeyRef:
              name: RELEASE-NAME-fernet-key
              key: fernet-key
        - name: AIRFLOW__CORE__SQL_ALCHEMY_CONN
          valueFrom:
            secretKeyRef:
              name: RELEASE-NAME-airflow-metadata
              key: connection
        - name: AIRFLOW_CONN_AIRFLOW_DB
          valueFrom:
            secretKeyRef:
              name: RELEASE-NAME-airflow-metadata
              key: connection
      envFrom: []
      image: python:3.9.7
      imagePullPolicy: IfNotPresent
      name: base
      ports: []
      volumeMounts:
        - mountPath: /opt/airflow/logs
          name: airflow-logs
        - mountPath: /opt/airflow/dags
          name: airflow-dags
          subPath: repo/airflow/example_dags
          readOnly: false
  hostNetwork: false
  restartPolicy: Never
  securityContext:
    runAsUser: 50000
    fsGroup: 50000
  nodeSelector: {}
  affinity: {}
  tolerations: []
  serviceAccountName: RELEASE-NAME-worker-serviceaccount
  volumes:
    - name: airflow-dags
      emptyDir: {}
    - name: airflow-logs
      emptyDir: {}
    - configMap:
        name: RELEASE-NAME-airflow-config
      name: airflow-config
    - configMap:
        name: RELEASE-NAME-airflow-config
      name: airflow-local-settings

My issues:
===========================================
I m deploying the k8sExecutor with in_cluster = False and use airflow_pod_template.yml  to config the executor pod
when I trigger the DAG, I found that the executor pod in k8s can not boot up , loop for failure as pic:
<img width="1440" alt="Screen Shot 2021-10-14 at 2 25 11 AM" src="https://user-images.githubusercontent.com/9009522/137192704-94b2c74c-2326-44ef-97c8-4c4f07f31b84.png">

I do put the image as python:3.9.7, but it show ":" only, don't know why?
Is there i missing change some config in airflow.cfg or my airflow_pod_template.yml was wrong, anybody can help me, pls?

**Comment by hterik** [comment]
Make sure AIRFLOW__KUBERNETES__WORKER_CONTAINER_REPOSITORY and AIRFLOW__KUBERNETES__WORKER_CONTAINER_TAG are set. They substitute whatever is set as image in the pod_template_file.
